python/taichi/ui/utils.py

```diff
@@ -64,8 +67,74 @@ def vec_to_euler(v):
     return yaw, pitch
 
 
+def try_get_wheel_tag(module):
+    try:
+        import wheel.metadata
+        wheel_path = f'{module.__path__[0]}-{".".join(map(str, module.__version__))}.dist-info/WHEEL'
+        meta = wheel.metadata.read_pkg_info(wheel_path)
+        return meta.get('Tag')
+    except Exception:
+        pass
+
+
+def try_get_loaded_libc_version():
+    assert platform.system() == "Linux"
+    with open('/proc/self/maps') as f:
+        content = f.read()
+
+    try:
+        import re
+        libc_path = next(v for v in content.split() if 'libc-' in v)
+        ver = re.findall(r'\d+\.\d+', libc_path)
+        if not ver:
+            return None
+        return tuple([int(v) for v in ver[0].split('.')])
+    except StopIteration:
+        return None
+
+
+class GGUINotAvailableException(Exception):
+    pass
+
+
 def check_ggui_availability():
     """Checks if the `GGUI` environment is available.
     """
-    if not _ti_core.GGUI_AVAILABLE:
-        raise Exception("GGUI Not Available")
+    if _ti_core.GGUI_AVAILABLE:
+        return
+
+    try:
+        # Try identifing the reason
+        wheel_tag = try_get_wheel_tag(taichi)
+        if platform.system(
+        ) == "Linux" and wheel_tag and 'manylinux2014' in wheel_tag:
+            libc_ver = try_get_loaded_libc_version()
+            if libc_ver and libc_ver < (2, 27):
+                raise GGUINotAvailableException(
+                    "GGUI is not available since you have installed a restricted version of taichi. "
+                    "Your OS is outdated, try upgrading to a recent one (e.g. Ubuntu 18.04 or later) if possible."
+                )
+
+            try:
+                import pip
+                ver = tuple([int(v) for v in pip.__version__.split('.')])
+                if ver < (20, 3, 0):
+                    raise GGUINotAvailableException(
+                        "GGUI is not available since you have installed a restricted version of taichi. "
+                        "Your pip (version {}) is outdated (20.3.0 or later required), "
+                        "try upgrading pip and install taichi again.".format(
+                            pip.__version__))
+            except ImportError:
+                pass
+
+            raise GGUINotAvailableException(
+                "GGUI is not available since you have installed a restricted version of taichi."
+            )
+
+    except GGUINotAvailableException:
+        raise
+
+    except Exception:
+        pass
```

Comment:
OOC: when do we expect to hit this? ;)

Reply:
When the code is buggy :)
This is a defending mechanism, if the code fails, just fallback to the most general situation, or user will see some confusing errors.

Reply:
Aha sg!